common/persistence/persistence_metric_clients.go

```diff
@@ -338,7 +339,13 @@ func (p *executionPersistenceClient) GetCurrentExecution(
 		p.healthSignals.Record(request.ShardID, caller, time.Since(startTime), retErr)
 		p.recordRequestMetrics(metrics.PersistenceGetCurrentExecutionScope, caller, time.Since(startTime), retErr)
 	}()
-	return p.persistence.GetCurrentExecution(ctx, request)
+	execution, retErr := p.persistence.GetCurrentExecution(ctx, request)
+	var notFound *serviceerror.NotFound
+	if errors.As(retErr, &notFound) {
+		// strip persistence-specific error message
+		retErr = serviceerror.NewNotFound("workflow not found")
+	}
```

Comment:
Initially, I had a fix in the persistence-specific implementations; but this seems easier and safer (as it covers any current and future ones).